packages/wallet/src/redux/protocols/transaction-submission/actions.ts

```diff
@@ -1,52 +1,60 @@
+import { WalletProtocol } from '../../types';
+import { BaseProcessAction } from '../actions';
+
 export type TransactionAction =
   | TransactionSentToMetamask
   | TransactionSubmissionFailed
   | TransactionSubmitted
   | TransactionConfirmed
   | RetryTransaction;
 
-export const TRANSACTION_SENT_TO_METAMASK = 'WALLET.COMMON.TRANSACTION_SENT_TO_METAMASK';
-export const TRANSACTION_SUBMISSION_FAILED = 'WALLET.COMMON.TRANSACTION_SUBMISSION_FAILED';
-export const TRANSACTION_SUBMITTED = 'WALLET.COMMON.TRANSACTION_SUBMITTED';
-export const TRANSACTION_CONFIRMED = 'WALLET.COMMON.TRANSACTION_CONFIRMED';
-export const RETRY_TRANSACTION = 'WALLET.COMMON.RETRY_TRANSACTION';
+export const TRANSACTION_SENT_TO_METAMASK = 'WALLET.TRANSACTION_SENT_TO_METAMASK';
+export const TRANSACTION_SUBMISSION_FAILED = 'WALLET.TRANSACTION_SUBMISSION_FAILED';
+export const TRANSACTION_SUBMITTED = 'WALLET.TRANSACTION_SUBMITTED';
+export const TRANSACTION_CONFIRMED = 'WALLET.TRANSACTION_CONFIRMED';
+export const RETRY_TRANSACTION = 'WALLET.RETRY_TRANSACTION';
+export const TRANSACTION_FINALIZED = 'WALLET.TRANSACTION_FINALIZED';
 
-export interface TransactionSentToMetamask {
+export interface TransactionSentToMetamask extends BaseProcessAction {
   type: typeof TRANSACTION_SENT_TO_METAMASK;
+  protocol: WalletProtocol.TransactionSubmission;
   channelId: string;
   processId: string;
 }
-export interface TransactionSubmissionFailed {
+export interface TransactionSubmissionFailed extends BaseProcessAction {
```

Comment:
On reading `BaseProcessAction` I see that is has a single property: `processId`. It makes sense to me that `TransactionSubmissionFailed` would _implement_ `BaseProcessAction` (i.e. conform to its interface) but it doesn't need to extend `BaseProcessAction` to do this - in fact it already has this property below. I don't think the tradeoff of having to look up another interface in another file makes sense in this case.

Reply:
It doesn't make sense for an interface to _implement_ another interface. An interface is a contract -- classes can implement a contract, and interfaces can extend on other contracts.

I don't really understand the resistance towards extending interfaces as a means of documentation and clarity. "A process action needs to have a `processId`" is a valid thing to be explicit about. Of course, you can be explicit about that in the definition of a `isProcessAction` (naming?) typeguard. However, that's a different issue than being explicit about the fact that `TransactionSubmissionFailed` _needs to be a process action_. That's the purpose of extending a common interface.

Without this, the connection between the `TransactionSubmissionFailed` interface and the definition of what it means to be a process action is lost: there is no connection made in the codebase, only in the head of a developer that manages to keep in their head the definition of `isProcessAction`, at all times.

Keeping the redundant `processId` property was in an effort to make it easier to see what is on an object of type `TransactionSubmissionFailed`, like you want to do by not extending something like the `BaseProcessAction`.

IMO, this is not great: if you want to rename `processId` to `protocolLocator`, for instance, it's so much easier to do so if every process action simply extends a `BaseProcessAction` interface. Then you rename it in two places: once in the base interface, and once in the type guard. The current approach lets you get the effect you want with just those two renames, with the negative side effect that you end up with a `processId` on all of these interfaces that you wanted to be explicit about. To me, that still seems better than just having `processId` on everything, with no common definition of what a "process action is" -- then, you have to search and replace across the codebase and hope you don't make a mistake.